### PR TITLE
Replace Config::Options#delegate to simple method definitions

### LINKED ISSUE #184: Undefined method 'delegate' when used with Padrino 0.14
I'm trying to use `Config` with Padrino 0.14. 

I've added the entry to the Gemfile and added the `register Config` line to `app.rb`. 

However when I try and start Padrino, I get the following exception:
```
/Users/gavinw/.rvm/gems/ruby-2.3.1@global/gems/bundler-1.15.0/lib/bundler/runtime.rb:85:in `rescue in block (2 levels) in require': There was an error while trying to load the gem 'config'. (Bundler::GemRequireError)
Gem Load Error is: undefined method `delegate' for Config::Options:Class
Did you mean?  DelegateClass
Backtrace for gem load error is:
/Users/gavinw/.rvm/gems/ruby-2.3.1@padrino/gems/config-1.6.0/lib/config/options.rb:156:in `<class:Options>'
/Users/gavinw/.rvm/gems/ruby-2.3.1@padrino/gems/config-1.6.0/lib/config/options.rb:5:in `<module:Config>'
/Users/gavinw/.rvm/gems/ruby-2.3.1@padrino/gems/config-1.6.0/lib/config/options.rb:4:in `<top (required)>'
/Users/gavinw/.rvm/gems/ruby-2.3.1@padrino/gems/config-1.6.0/lib/config.rb:4:in `require'
/Users/gavinw/.rvm/gems/ruby-2.3.1@padrino/gems/config-1.6.0/lib/config.rb:4:in `<top (required)>'
/Users/gavinw/.rvm/gems/ruby-2.3.1@global/gems/bundler-1.15.0/lib/bundler/runtime.rb:82:in `require'

```

Any ideas?

### PR DESCRIPTION
**Pull request by yhatt**
Fix #184.

`Config::Options#delegate` is using ActiveSupport's `#delegate` implicitly. So it would raise
`NoMethodError` with running in not rails environment. e.g. Padrino, Sinatra and pure Ruby.

It could fix it by replacing to simple method definitions.

### CODE REVIEW

**Comment by pkuczynski** [comment]
Thanks for the fix. Could you please add a test for it?

**Comment by yhatt** [comment]
Well, we need to add not-rails environment to test it. I'll try writing by Sinatra.

**Comment by pkuczynski** [comment]
Cool, thanks! That should be easily possible with Appraisals

File: spec/spec_helper.rb [hunk 1]
```diff
@@ -14,27 +14,55 @@
...
+  require File.expand_path("../app/#{app_name}/app", __FILE__)
+
+  # Load Rspec
+  require 'rspec'
+
+  # Configure
+  RSpec.configure do |config|
+    config.filter_run_excluding :rails
 
+    def fixture_path
```
Comment (yhatt) [inline spec/spec_helper.rb]: For preventing to fail current specs, I defined `fixture_path` manually in sinatra test. `fixture_path` is provided by rspec-rails.

**Comment by yhatt** [comment]
Yay, checks passed! Adding sinatra test was a bit harder than I thought 😅

**Comment by pkuczynski** [comment]
You did a great job! :) I am waiting for Travis to report the build status and we can merge it in :)

**Comment by pkuczynski** [comment]
All good! Merging in and releasing...

**Pull request merged**